health/health.go

```diff
@@ -840,13 +842,19 @@ func (t *Tracker) updateBuiltinWarnablesLocked() {
 					ArgCurrentVersion:   version.Short(),
 					ArgAvailableVersion: cv.LatestVersion,
 				})
-			} else {
+			} else if !t.applyUpdates.EqualBool(true) { // only show update warning when auto-updates are off
 				t.setUnhealthyLocked(updateAvailableWarnable, Args{
 					ArgCurrentVersion:   version.Short(),
 					ArgAvailableVersion: cv.LatestVersion,
 				})
+			} else {
+				t.setHealthyLocked(updateAvailableWarnable)
 			}
+		} else {
+			t.setHealthyLocked(updateAvailableWarnable)
 		}
+	} else {
+		t.setHealthyLocked(updateAvailableWarnable)
```

Comment:
You could probably pull this out into a helper func that prevents needing these layers of repeated code at different indent layers.

And/or all this health.Tracker stuff is finally nicely testable now finally without global gunk, so a table test for these is no longer tedious.